cpp/include/raft/cuda_utils.cuh

```diff
@@ -109,7 +109,7 @@ static const int WarpSize = 32;
 DI int laneId()
 {
   int id;
-  asm("mov.s32 %0, %laneid;" : "=r"(id));
+  asm("mov.s32 %0, %%laneid;" : "=r"(id));
```

Comment:
Is there some sort of documentation or guidelines we can add to our new [developer guide](https://github.com/rapidsai/raft/blob/branch-22.02/DEVELOPER_GUIDE.md) which would make it easier for developers to follow these rules in the future?

Reply:
It's part of the inline PTX assembly spec: https://docs.nvidia.com/cuda/inline-ptx-assembly/index.html#parameters (quote : "If you want the `%` in a ptx instruction, then you should escape it with double `%%`, e.g.: [...] ").
I honestly have no idea why `nvcc` even compiles the above code, I guess they have added some special cases for the special registers against their own docs.
I can add a link to the spec in the developer guide, but IMO, this is more confusing than anything. The underlying issue is that `nvcc` accepts this (clearly incorrect) code.